coriolis/tests/test_context.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+from coriolis import context
+from coriolis.tests import test_base
+from coriolis.tests import testutils
+
+
+class RequestContextTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis RequestContext class."""
+
+    def setUp(self):
+        super(RequestContextTestCase, self).setUp()
+        self.req_context = context.RequestContext(
+            user="user",
+            project_id='test_project_id',
+        )
+
+    @mock.patch.object(context.timeutils, 'utcnow')
+    def test__init__(self, mock_utcnow):
+        context_req = testutils.get_wrapped_function(context.RequestContext)(
+            user="user",
+            project_id='test_project_id',
+            roles=['role1', 'role2'],
+        )
+
+        mock_utcnow.assert_called_once_with()
+        self.assertEqual(context_req.timestamp, mock_utcnow.return_value)
+        self.assertEqual(context_req.roles, ['role1', 'role2'])
+
+    @mock.patch.object(context.timeutils, 'parse_isotime')
+    def test__init__with_string_timestamp(self, mock_parse_isotime):
+        new_context = testutils.get_wrapped_function(context.RequestContext)(
+            user="user",
+            project_id='test_project_id',
+            timestamp="2023-11-09T13:31:08Z",
+        )
+
+        mock_parse_isotime.assert_called_once_with("2023-11-09T13:31:08Z")
+        self.assertEqual(new_context.timestamp,
+                         mock_parse_isotime.return_value)
+        self.assertEqual(new_context.roles, [])
+
+    def test_to_dict(self):
+        result = self.req_context.to_dict()
+
+        self.assertEqual(result['timestamp'],
+                         self.req_context.timestamp.isoformat())
+
+    def test_from_dict(self):
+        result = self.req_context.to_dict()
```

Comment:
Here you don't need to check whether the timestamp is handled, you already do that in your `__init__` test. I think you should either:
1. Check that the class is called with `**result`
2. Remove the test case altogether. It's fine if `from_dict` is not covered, it doesn't really do much.

Reply:
Gotcha :) Thank you!